specviz/plugins/model_editor/model_editor.py

```diff
@@ -192,17 +192,19 @@ def _redraw_model(self):
                 spectrum = spectrum.new_flux_unit(plot_data_item.data_unit)
                 plot_data_item.data_item.set_data(spectrum)
 
-        plot_data_item.set_data()
+            # Only draw if ModelDataItem
+            plot_data_item.set_data()
 
     def _on_model_item_changed(self, item):
         if item.parent():
             # If the item has a parent, then we know that the parameter
             # value has changed. Note that the internal stored data has not
             # been truncated at all, only the displayed text value. All fitting
             # uses the full, un-truncated data value.
-            item.setData(float(item.text()), Qt.UserRole + 1)
-            # item.setText("{:.5g}".format(float(item.text()))) # dont change user input
-            item.setText(item.text())
+            if item.column() == 1:
+                item.setData(float(item.text()), Qt.UserRole + 1)
+                # item.setText("{:.5g}".format(float(item.text()))) # dont change user input
```

Comment:
Should remove this comment if it's no longer useful.

Reply:
This line is not final. There needs to be a discussion on how to display the fitted params.